implementation/libs/jbb-lib-cache/src/main/java/org/jbb/lib/cache/ProxySpringCacheManager.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2017 the original author or authors.
+ *
+ * This file is part of jBB Application Project.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may obtain a copy of the License at
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.jbb.lib.cache;
+
+import org.springframework.cache.CacheManager;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Delegate;
+
+
+@Getter
+@Setter
+class ProxySpringCacheManager implements CacheManager {
```

Comment:
It's all about reloading some beans at runtime. Reloading single bean is hard because it is injected to another beans, they need to refresh also etc.

So, workaround is to use some proxy class which is injected everywhere. Proxy has a real cache manager inside due to Delegate annotation taken from lombok library.